---
title: "Analysis Report 1: Forensic Identification using single taxa distribution of skin bacterial communities "
author: "Maggie Chen"
date: "October 20, 2017"
output: github_document
bibliography: references.bib
csl: bioinformatics.csl
---

```{r setup, include=FALSE, echo=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

# Introduction
Bacteria community constantly interacts with human communities, and it has both positive and negative impact on humans. for instance: benifits and contrast. skin disease. 
However, human skin is also good carrier of bacteria, and transfer them onto surfaces through touching. Also, individal's living environment and life style imapcts the bacteria community on each person, which creates a significan variations between individals (CITE). What's more, skin bacteria communities are relative stable over times (CITE8). In addition, bacteria is capable to persist on the touched surface for certain amout of time, even in spite of temperature and storage conditions(CITE). Along with the fast development of sequencing techniques, the sequence and investigation on microbial communities has expanded extensively. For example, (Sequencning technology, DADA) In addition to the health implication of understand bacteria, another relative new application of understanding microbe community is in the forensic examinations. The combination of the high throughput sequencing technology and the characterristics of bactieria communities on human skins and how they persist on touched surface makes it possible as a forensic identification method. Through comparing and matching the bacteria community composition, the microbal community will serve as an "fingerprint" for the individal identification during forensic incestigations (CITE).

The goal for this article, is to explore additional approach of analyze mriobial community for forensic identification surposes. My primary questuion is that instead of matching multiple communities between human skin and object been touched, how single taxa analysis can help with forensic identification. My hypothesis is that in the male community, for the most abundant bactiera taxa will has unique distribution on object, in this case, the keyboards. This unique bacteria distribution will then function as an "fingerprint" for individual identification. In order investigate on this hypothesis, R prgram is used for sequence analysis.  
 
Recent research has indicated that microbial community can serve as an additional evidence in forensic investigations. Using the high throughput sequencing technologies to identify unique microbial community on individuals. In addition to match the microbial community composition, which contains multiple taxa. It is possible to use the distribution and abundance of one taxa on different parts of objects as an individualized fingerprint‚Äù. This new approach will provides an indepth evidence during forensic investigation.

Add about 1.5-2 pages here. Must cite at least 5 peer reviewed articles.

# Methods

## Sample origin and sequencing

Add about half a page here. In this section instead of first person (I/we), use Fierer et al., since you'll just be describing what they did, based on the methods in their paper.

## Computational

These are the methods you used. Should probably be at least a half of a page. At a very minimum should include citations for DADA2 [@callahan2016] and phyloseq [@mcmurdie2013]. Note that these don't count towards the five references you need to cite in the introduction.

# Results

In addition to a minimum of 3-4 figures/tables (and associated captions), you should include sufficient text in this section to describe what your findings were. Remember that in the results section you just describe what you found, but you don't interpret it - that happens in the discussion.

```{r load-libraries, message = FALSE}
# Be sure to install these packages before running this script
# They can be installed either with the intall.packages() function
# or with the 'Packages' pane in RStudio

# load general-use packages
library("dplyr")
library("tidyr")
library("knitr")
library("ggplot2")

# this package allows for the easy inclusion of literature citations in our Rmd
# more info here: https://github.com/crsh/citr
# and here:
# http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html
library("citr")

# These are the primary packages well use to clean and analyze the data
# this package needs to be installed from bioconductor -- it's not on CRAN
# see info here: https://benjjneb.github.io/dada2/dada-installation.html
library("dada2")

# This to export a fasta of our final denoised sequence variants
library("seqinr")

# To install this you have to install from GitHub
# See more info here: https://github.com/leffj/mctoolsr
# run this -- install.packages("devtools")
# and then this -- devtools::install_github("leffj/mctoolsr")
library("mctoolsr")

# And this to visualize our results
# it also needs to be installed from bioconductor
library("phyloseq")
```

```{r extract-sample-and-file-names}
# NOTE: Much of the following follows the DADA2 tutorials available here:
# https://benjjneb.github.io/dada2/tutorial.html
# Accessed October 19, 2017

# set the base path for our input data files
path <- "data/raw_data"

# Sort ensures samples are in order
filenames_forward_reads <- sort(list.files(path, pattern = ".fastq"))

# Extract sample names, assuming filenames have format: SAMPLENAME.fastq
sample_names <- sapply(strsplit(filenames_forward_reads, "\\."), `[`, 1)

# Specify the full path to each of the filenames_forward_reads
filenames_forward_reads <- file.path(path, filenames_forward_reads)
```

```{r check-quality-plots, include = FALSE}
# Plots the quality profiles of all twenty samples
plotQualityProfile(filenames_forward_reads[1:115])
```

We can see from the quality profiles that most reads tend to get pretty bad in quality after around 200 bases. 

```{r filter-reads, include = FALSE}
# Place filtered files in filtered/ subdirectory
# note this will fail if the directory doesn't exist
filter_path <- file.path("output", "filtered")
filtered_reads_path <- file.path(filter_path,
                                 paste0(sample_names,
                                        "_filt.fastq.gz"))

# See ?filterAndTrim for details on the parameters
# See here for adjustments for 454 data:
# https://benjjneb.github.io/dada2/
#     faq.html#can-i-use-dada2-with-my-454-or-ion-torrent-data
filtered_output <- filterAndTrim(fwd = filenames_forward_reads,
                                 filt = filtered_reads_path,
                                 maxLen = 300,
                                 maxN = 0, # discard any seqs with Ns
                                 maxEE = 3, # allow w/ up to 3 expected errors
                                 truncQ = 2, # cut off if quality gets this low
                                 rm.phix = TRUE,
                                 compress = TRUE,
                                 multithread = FALSE)
```

```{r filtered-read-counts-table, include = FALSE}
# produce nicely-formatted markdown table of read counts
# before/after trimming
kable(filtered_output,
      col.names = c("Reads In",
                    "Reads Out"))
```

```{r learn-errors, include = FALSE}
# this build error models from each of the samples
errors_forward_reads <- learnErrors(filtered_reads_path,
                                    multithread = FALSE)
```

```{r visualize-errors-with-plots,include = FALSE}
# quick check to see if error models match data
# (black lines match black points) and are generally decresing left to right
plotErrors(errors_forward_reads,
           nominalQ = TRUE)
```

```{r dereplicate-sequences, include = FALSE}
# get rid of any duplicated sequences
dereplicated_forward_reads <- derepFastq(filtered_reads_path,
                                         verbose = TRUE)

# Name the derep-class objects by the sample names
names(dereplicated_forward_reads) <- sample_names
```

```{r run-dada, include = FALSE}
# parameters adjusted based on recommendations for 454 data here:
# https://benjjneb.github.io/dada2/
#     faq.html#can-i-use-dada2-with-my-454-or-ion-torrent-data
dada_forward_reads <- dada(dereplicated_forward_reads,
                           err = errors_forward_reads,
                           HOMOPOLYMER_GAP_PENALTY = -1, # reduce penalty bc 454
                           BAND_SIZE = 32) # performs local alignments bc indels

# check dada results
dada_forward_reads
```

```{r make-sequence-table, include = FALSE}
# produce the 'site by species matrix'
sequence_table <- makeSequenceTable(dada_forward_reads)
```

The output table has `r nrow(sequence_table)` rows (samples) and `r ncol(sequence_table)` columns (sequence variants). Notice how we can embed R code directly in our markdown text.

```{r histogram-of-sequence-lengths, include = FALSE}
# Quick check to look at distribution of trimmed and denoised sequences
hist(nchar(getSequences(sequence_table)),
     main = "Histogram of final sequence variant lengths",
     xlab = "Sequence length in bp")
```

```{r remove-chimeras, include = FALSE}
# Check for and remove chimeras
sequence_table_nochim <- removeBimeraDenovo(sequence_table,
                                            method = "consensus",
                                            multithread = FALSE,
                                            verbose = TRUE)

# What percent of our reads are non-chimeric?
non_chimeric_reads <- round(sum(sequence_table_nochim) / sum(sequence_table),
                            digits = 4) * 100
```

After removing chimeras, we were left with `r non_chimeric_reads`% of our cleaned reads.

```{r table-of-pipeline-read-counts, include = FALSE}
# Build a table showing how many sequences remain at each step of the pipeline
get_n <- function(x) sum(getUniques(x)) # make a quick function
track <- cbind(filtered_output, # already has 2 columns
               sapply(dada_forward_reads, get_n),
               rowSums(sequence_table),
               rowSums(sequence_table_nochim))

# add nice meaningful column names
colnames(track) <- c("Input",
                     "Filtered",
                     "Denoised",
                     "Sequence Table",
                     "Non-chimeric")

# set the proper rownames
rownames(track) <- sample_names

# produce nice markdown table of progress through the pipeline
kable(track)
```

```{r assign-taxonomy, include = FALSE}
# assigns taxonomy to each sequence variant based on a supplied training set
# made up of known sequences
taxa <- assignTaxonomy(sequence_table_nochim,
                       "data/training/rdp_train_set_16.fa.gz",
                       multithread = FALSE,
                       tryRC = TRUE) # also check with seq reverse compliments

# show the results of the taxonomy assignment
unname(taxa)
```

```{r extract-sequences-to-fasta}
# we want to export the cleaned, trimmed, filtered, denoised sequence variants
# so that we can build a phylogeny - we'll build the phylogeny outside of R
# but we need the fasta file to do so. We keep the names of each sequence as the
# sequence itself (which is rather confusing), because that's how DADA2 labels
# it's columns (e.g. 'species')
# function taken from https://github.com/benjjneb/dada2/issues/88
export_taxa_table_and_seqs <- function(sequence_table_nochim,
                                       file_seqtab,
                                       file_seqs) {
  seqtab_t <- as.data.frame(t(sequence_table_nochim)) # transpose to data frame
  seqs <- row.names(seqtab_t) # extract rownames
  row.names(seqtab_t) <- seqs # set rownames to sequences
  outlist <- list(data_loaded = seqtab_t)
  mctoolsr::export_taxa_table(outlist, file_seqtab) # write out an OTU table
  seqs <- as.list(seqs)
  seqinr::write.fasta(seqs, row.names(seqtab_t), file_seqs) # write out fasta
}

# actually run the function, with the names of the files we want it to create
# and where to put them
export_taxa_table_and_seqs(sequence_table_nochim,
                           "output/sequence_variants_table.txt",
                           "output/sequence_variants_seqs.fa")
```


```{r read-in-metadata-and-create-phyloseq}
# Next we want to read in the metadata file so we can add that in too
# This is not a csv file, so we have to use a slightly different syntax
# here the `sep = "\t"` tells the function that the data are tab-delimited
# and the `stringsAsFactors = FALSE` tells it not to assume that things are
# categorical variables
metadata_in <- read.table(paste0("data/metadata/",
                    "fierer_hand_bacteria_SRA_study_ERP022626_SraRunTable.txt"),
                          sep = "\t",
                          header = TRUE,
                          stringsAsFactors = FALSE,
                          row.names = 6) # sets sample IDs to row names

# read in the phylogeny, which was created from the fasta exported above
# in Geneious by aligning the sequences with MAFFT and then building a
# Maximum-Likelihood tree with RAxML
tree_in <- read_tree("output/sequence_variants_MAFFT_FastTree.newick")

# Construct phyloseq object (straightforward from dada2 outputs)
phyloseq_obj <- phyloseq(otu_table(sequence_table_nochim,
                                   taxa_are_rows = FALSE), # sample-spp matrix
                         sample_data(metadata_in), # metadata for each sample
                         tax_table(taxa), # taxonomy for each sequence variant
                         phy_tree(tree_in)) # phylogeny from sequence variants
```



```{r, combine_tables}
# Combine the taxa and metedata into one table
# Enable ggplot function to use both tables information at the same time
melted_obj <- psmelt(phyloseq_obj)
```

```{r, host-phylum-abuandace-barplot}
# Phylum distribution for each individual
# Also, in sepeartion of sample types
  melted_obj %>%
  ggplot(aes(x = host_subject_id,
             y = Abundance,
             fill = Phylum, color = Phylum)) +
  geom_col(position = position_dodge()) +
  facet_grid(sample_type ~.) +
  ggtitle("Abundance of Phylums Found on Individuals")

```

**Figure1**: Abundace of taxas found on each indivual and their keyboard samples, colored by Phylums. Seperate by sample types, skin as the fingertip and surface as the keyboard. As the Figure presents, the most commonly found Phylum across all the indivudals' keyboard samples is *Actinobacteria*, present in organge color. The second commnoly found bacteria Phylum is *Firmicutes*. This is consistant with fingertip samples, however, fingertip samples are only avaliable for M2, M3, M9 individals, who are all males. Other individuals does not have fingertip sample avaliable, including F1 who is a female individual; for L1, L3, R1, U1, U2, U3 the sex of the individual is not applictable.


```{r, Actinobacteria-Firmicutes_key_identification-barplot}
# Filter the most commonly found species, Actinobacteria and Firmicutes
# and examine the sample source of where Actinobacteria and Firmicutes was found
melted_obj %>%
  filter(Phylum %in% c("Actinobacteria", "Firmicutes")) %>%
  ggplot(aes(x = host_subject_id,
             y = Abundance,
             fill = sample_source,
             color = sample_source)) +
  geom_col(position = position_dodge()) +
  facet_grid(Phylum ~.) +
  ggtitle("Abundance of Actinobacteria and Firmicutes Found on Keys")

```

**Figure2**: Abundace of two most commnoly found bacteria phylums on each individuals, seperated by Phlums Actinobacteria and Firmicutes, colored by sample source. The sample source include each of the keys on the keyboard and individual fingertip samples. As an result, only male individuals M2, M3, M9 has all keys sampled. For other individuals, only space bar is sampled. The highest amount of Actinobacteria is found on figertip.



```{r, male-sample_type_Family-barplot}
# To examine for male indivudals phylums of Actinobacteria
# then look for the Family distribution on each keys
# for each of the male samples
melted_obj %>%
  filter(host_subject_id %in% c("M2", "M3", "M9")) %>%
  filter(Phylum %in% c("Actinobacteria")) %>%
  ggplot(aes(x = Family,
             y = Abundance,
             color = sample_source,
             fill = sample_source)) +
    geom_col(position = position_dodge()) +
  facet_grid(host_subject_id ~.) +
  theme(axis.text.x = element_text(angle = 90,
                                    hjust = 1)) +
  ggtitle("Abundance of Families found on Different Keys
          for M2, M3, M9")

```

**Figure3**: Abundance of Families within phylum Actinobacteria on M2, M3, M9 individuals, seperate by individuals, and colored by sample source. The sample source include each of the keys on the keyboard and individual fingertip samples. Propionibacteriaceae family has a significant higher abundance in all individual samlple across all keys and fingertips. For M3 and M9, fingertip is the location that has most of Propionibacteriaceae found. For M2, all keys and fingertip has relatie lower abundace than M3 and M9. All three individuals have different distuributions of Propionibacteriaceae on keys. 


```{r, Genus-in-Propionibacteriaceae_Family-barplot}
# Filter the Family Propionibacteriaceae, which is most common
# Further look for Genus distribution in 3 male samples
# on each of the keys examed
melted_obj %>%
  filter(host_subject_id %in% c("M2", "M3", "M9")) %>%
  filter(Family == "Propionibacteriaceae") %>%
  ggplot(aes(x = Genus,
             y = Abundance,
             color = sample_source,
             fill = sample_source)) +
    geom_col(position = position_dodge()) +
    facet_grid(host_subject_id ~.) +
  theme(axis.text.x = element_text(angle = 90,
                                    hjust = 1)) +
  ggtitle("Abundance of Genus in Propionibacteriaceae
          family on M2, M3, M9")

```

**Figure4**: Genus abundance in Propionibacteriaceae family on M2, M3, M9. For all three male individuals M2, M3, M9, the most common Genus identified on both their keyboards and fingertip is Propionibacterium. For individual M3 and M9, the most adundance on found on Fingertip. For each of the individual, there is a different distrubution of amount of Genus Propionibacterium found on each of the keys.

```{r, subset-Genus}
# use subset data to generate a selected dataset
# for the alpha diversity check
# create a subset for only Genus Propionibacterium
subset_genus <- subset_taxa(phyloseq_obj, Genus == "Propionibacterium")

```

```{r, subset-Male}
# use subset data to generate a selected dataset
# for the alpha diversity check
# make another subset that only contains information
# of Genus Propionibacterium for M2,3,9
subset2_male <- subset_samples(subset_genus,
                               host_subject_id %in% c("M2", "M3", "M9"))
```

```{r, jitterplot-sample-diversity}
# Use created subset data to plot for the
# alpha diversity metirc on different keys
plot_richness(subset2_male,
              x = "sample_type",
              measures = c("Shannon"),
              color = "sample_source") +
  xlab("Sample origin") +
  geom_jitter(width = 0.2) +
  facet_grid(host_subject_id ~.) +
  theme_bw() +
  ggtitle("Alpha Diversity of Genus Propionibacterium on M2, M3, M9 Samples")
```

**Figure5**: The male sample alpha diversity of sequences identified as Genus *Propionibacterium*, colored by sample source. This figure presents the alpha diveristy on sample from M2, M3, and M9, which are all male individuals. The alpha diverity id measured for the sequences identified as Genus *Propionibacterium*. The samples are from both fingertip and the keyboard of the individual. The M2 has a higher diversity in this geneus. And the fingertip in M2 has higher diversity than its keyboard. For M3 and M9, they have lower diveristy in the samples compare to M2. Also, for both M3 and M9, the fingertip and keyboards has a very simmliar diversities. 





# Discussion

The higer alpha diversity in the M2 in dicate the possibleity of have different strains or different species that is hard for sequence to detect

Add around 2-3 pages interpreting your results and considering future directions one might take in analyzing these data.

# Sources Cited


